src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,109 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs an empty list of tasks.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a task list containing the elements of the specified ArrayList of Tasks,
+     * in the order they are returned by the list's iterator.
+     *
+     * @param tasks the ArrayList of Tasks whose tasks are to be placed into this task list
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Returns the task at the specified position in this task list.
+     *
+     * @param index the 1-based index of the task to return
+     * @return the task at the specified position in this task list
+     * @throws DukeException if the index is out of range (index < 0 || index >= size())
+     */
+    public Task get(int index) throws DukeException {
```

Comment:
I think using a more specific naming like getAtPosition(int index) would be clearer.